### ISSUE TITLE
TLS configuration

### ISSUE THREAD

**Issue by mateivladimir** [pgbackrest/pgbackrest#2317]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: v 2.50

2. PostgreSQL version: 15.5

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: a database server, a "control" server (just for pgbackrest command) and a MiniO server

4. Did you install pgBackRest from source or from a package? from source

5. Please attach the following as applicable:
    - `pgbackrest.conf` from DB server: 
[backups]
pg1-host=pg1-srv
pg1-path=/data/pgsql-15
pg1-database=postgres
pg1-host-type=tls
pg1-host-cert-file=/etc/macerts/backup-srv.crt
pg1-host-key-file=/etc/macerts/backup-srv.key
pg1-host-ca-file=/etc/macerts/ca.crt

[global]
tls-server-address=*
tls-server-cert-file=backup-srv.crt
tls-server-key-file=backup-srv.key
tls-server-ca-file=ca.crt
tls-server-auth=pg1-srv=backups

repo1-path=/value
repo1-type=s3
repo1-s3-endpoint=https://x.x.x.x
repo1-s3-verify-tls=n
repo1-s3-bucket=backup
repo1-s3-key=.....
repo1-s3-key-secret=...
repo1-s3-uri-style=path
repo1-s3-region=eu-west-2
repo1-retention-full=10

archive-async=y
process-max=4
log-level-console=debug
log-level-file=info
start-fast=y
delta=y

    - `pgbackrest.conf` from "controller" server:
[backups]
pg1-host=pg1-srv
pg1-path=/data/pgsql-15
pg1-database=postgres
pg1-host-type=tls

[global]
repo1-host-type=tls
repo1-host-cert-file=pg1-srv.crt
repo1-host-key-file=pg1-srv.key
repo1-host-ca-file=ca.crt

tls-server-address=*
tls-server-cert-file=pg1-srv.crt
tls-server-key-file=pg1-srv.key
tls-server-ca-file=ca.crt
tls-server-auth=backup-srv=demo

repo1-path=/value
repo1-type=s3
repo1-s3-endpoint=https://x.x.x.x
repo1-s3-verify-tls=n
repo1-s3-bucket=backup
repo1-s3-key=...
repo1-s3-key-secret=...
repo1-s3-uri-style=path
repo1-s3-region=eu-west-2
repo1-retention-full=1

archive-async=y
process-max=2
log-level-console=debug
log-level-file=info
start-fast=y
delta=y

7. Describe the issue: I've got the issue:

2024-04-08 12:57:43.915 P00  DEBUG:     protocol/helper::protocolRemoteParam: => {["--exec-id=1081759-614a41a9", "--log-level-console=off", "--log-level-file=off", "--log-level-stderr=error", "--pg1-database=postgres", "--pg1-path=/data/pgsql-15", "--process=0", "--remote-type=pg", "--stanza=backups", "check:remote"]}
WARN: unable to check pg1: [AccessError] raised from remote-0 tls protocol on 'pg1-srv': access denied
2024-04-08 12:57:43.918 P00  DEBUG:     db/helper::dbGet: => struct
2024-04-08 12:57:43.918 P00  DEBUG:     command/exit::exitSafe: (result: 0, error: true, signalType: 0)
ERROR: [027]: no database found
       HINT: check indexed pg-path/pg-host configurations

**Comment by pgstef** [comment]
What's the purpose of TLS connections in your use-case? I mean you're using S3 storage, why do you need a "controller" for? Can't you run the backup command on the db host directly?
The all point of using TLS connections is to have a dedicated backup server (repo-host, possibly storing things to s3 afterwards) and a local database.

> I know that the error is very clear, but i'm sure i have a database called "postgres" inside it

Well, you just totally messed-up your configuration. To fix it, you need to know what you want to do exactly.
See the [user guide](https://pgbackrest.org/user-guide-rhel.html#repo-host/config) for a detailed explanation of the TLS server setup.

**Comment by mateivladimir** [comment]
Yeah, indeed i totally messed-up my config. I try to configure TLS because I just want to remove 22 port between "controller" node and DB server. And the needed of "controller" node is to keep centralized all my backups and test if it's all ok only in that server. And also, if i want to change the S3 server, to change in 1 place instead of 10+ (or maybe more)

**Comment by pgstef** [comment]
Then I think you simply misunderstood the config location (and certs location) between your db and repo host.
Assuming `pg1-srv` is your db host and `backup-srv` is your backup server/controller (aka repo-host), and that your certs location is correct, I guess your configuration files could look like the following.

db host:

```ini
[global]
# repository details
repo1-host=backup-srv
repo1-host-type=tls
repo1-host-cert-file=pg1-srv.crt
repo1-host-key-file=pg1-srv.key
repo1-host-ca-file=ca.crt

# general options
archive-async=y
process-max=4
log-level-console=debug
log-level-file=info
delta=y

# tls server options
tls-server-address=*
tls-server-cert-file=pg1-srv.crt
tls-server-key-file=pg1-srv.key
tls-server-ca-file=ca.crt
tls-server-auth=backup-srv=backups

[backups]
# db location and access
pg1-path=/data/pgsql-15
```

repo host:

```ini
[global]
# repository details
repo1-path=/value
repo1-type=s3
repo1-s3-endpoint=https://x.x.x.x
repo1-s3-verify-tls=n
repo1-s3-bucket=backup
repo1-s3-key=...
repo1-s3-key-secret=...
repo1-s3-uri-style=path
repo1-s3-region=eu-west-2
repo1-retention-full=1

# general options
process-max=2
log-level-console=info
log-level-file=debug
start-fast=y
delta=y

# tls server options
tls-server-address=*
tls-server-cert-file=/etc/macerts/backup-srv.crt
tls-server-key-file=/etc/macerts/backup-srv.key
tls-server-ca-file=/etc/macerts/ca.crt
tls-server-auth=pg1-srv=backups

[backups]
# db location and access
pg1-host=pg1-srv
pg1-path=/data/pgsql-15
pg1-host-type=tls
pg1-host-cert-file=/etc/macerts/backup-srv.crt
pg1-host-key-file=/etc/macerts/backup-srv.key
pg1-host-ca-file=/etc/macerts/ca.crt
```

**Comment by mateivladimir** [comment]
Thanks for taking the time to test the setup. Until now, i didnt see that in controller server i've missed my "stanza" and i put "tls-server-auth=backup-srv=demo" instead of "tls-server-auth=backup-srv=backups". Many thanks again and I'm so sorry for taking your time.
